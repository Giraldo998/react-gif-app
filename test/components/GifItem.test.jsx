const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components/GifItem");
describe('Pruebas en GifItem', () => { 
   const title = 'saitama';
   const url = 'https://one-punch.com/saitama.jpg';

   test('Evaluar snapshot', () => {

      const {container} = render(<GifItem title={title} url={url}/>);
      expect(container).toMatchSnapshot();
      // screen.debug();
      // expect(screen.getByRole('img').src).toBe(url);
      const {src, alt} = screen.getByRole('img');
      expect(src).toBe(url)
      expect(alt).toBe(title)
   })

   test('Debe de mstrar el titulo en el componente', () => {
      render(<GifItem title={title} url={url}/>);
      expect(screen.getByText( title )).toBeTruthy();
    })
})


