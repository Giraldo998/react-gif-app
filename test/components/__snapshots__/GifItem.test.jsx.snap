// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Evaluar snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
